### PR TITLE
Requirements update

### PR DESCRIPTION
**Pull request by snopoke**
Keeping on staging for a while

### CODE REVIEW

**Comment by benrudolph** [comment]
:haircut: nice!

File: corehq/apps/hqadmin/system_info/checks.py [hunk 1]
```diff
@@ -54,32 +54,53 @@ def check_rabbitmq():
...
+                worker_stats = detailed_stats[worker_name]
+                running_tasks = worker_stats['pool']['writes']['inqueues']['active']
+                concurrency = worker_stats['pool']['max-concurrency']
+                completed_tasks = worker_stats['pool']['writes']['total']
+
+                tasks_class = tasks_full if running_tasks == concurrency else tasks_ok
+                tasks_html = mark_safe(
+                    '<span class="label %s">%d / %d</span> :: %d' % (
+                        tasks_class, running_tasks, concurrency, completed_tasks
+                ))
```
Comment (dimagimon) [inline corehq/apps/hqadmin/system_info/checks.py]: E122 continuation line missing indentation or outdented

File: corehq/apps/hqadmin/system_info/checks.py [hunk 2]
```diff
@@ -56,30 +56,57 @@ def check_rabbitmq():
...
-            pass
-        worker_ok = '<span class="label label-success">OK</span>'
-        worker_bad = '<span class="label label-important">Down</span>'
+            t = cresource.get(endpoint, params_dict=params).body_string()
+            return json.loads(t)
+        except Exception:
+            return {}
+
 
+    def get_task_html(detailed_stats, worker_name):
```
Comment (dimagimon) [inline corehq/apps/hqadmin/system_info/checks.py]: E303 too many blank lines (2)

File: corehq/apps/hqadmin/system_info/checks.py [hunk 3]
```diff
@@ -56,30 +56,57 @@ def check_rabbitmq():
...
+                    tasks_class, running_tasks, concurrency, completed_tasks
+                )
+            )
+        except KeyError:
+            pass
+
+        return tasks_html
+
+
+    ret = {}
```
Comment (dimagimon) [inline corehq/apps/hqadmin/system_info/checks.py]: E303 too many blank lines (2)

**Comment by benrudolph** [comment]
any thoughts on how close this is to merging? it conflicts with a lot of branches on staging, and while not super hard to resolve, does add to the complexity of merge conflicts when we have many `+requirements-update` branches. (don't want to rush it and cause bugs, just making sure it's tracked)

**Comment by snopoke** [comment]
It's been on staging for over a week now. I think its pretty safe. I'll do
a few more checks today and then sign off if nothing breaks.

On 28 October 2015 at 22:48, Ben <notifications@github.com> wrote:

> any thoughts on how close this is to merging? it conflicts with a lot of
> branches on staging, and while not super hard to resolve, does add to the
> complexity of merge conflicts when we have many +requirements-update
> branches. (don't want to rush it and cause bugs, just making sure it's
> tracked)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8793#issuecomment-151985862>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by benrudolph** [comment]
nice sounds good

**Comment by snopoke** [comment]
@benrudolph tested celery, reports, exports, downloads, pillowtop, api, form submissions
think this is GTG

**Comment by czue** [comment]
can you confirm you tested elastic reports? the `rawes` bump caught my eye

**Comment by snopoke** [comment]
I tested the case list and the worker activity report

On 30 October 2015 at 13:00, Cory Zue <notifications@github.com> wrote:

> can you confirm you tested elastic reports? the rawes bump caught my eye
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8793#issuecomment-152496145>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by czue** [comment]
:+1:

**Comment by czue** [comment]
:tada:

**Comment by snopoke** [comment]
nice, still more to do but can wait a bit

On 30 October 2015 at 13:53, Cory Zue <notifications@github.com> wrote:

> [image: :tada:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8793#issuecomment-152505160>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by benrudolph** [comment]
looks good!

**Pull request merged**